server/src/com/thoughtworks/go/server/service/UserService.java

```diff
@@ -131,6 +120,40 @@ private boolean userNameListContainsAdmin(List<String> enabledUserNames) {
         return false;
     }
 
+    public void update(User user, TriState enabled, TriState emailMe, String email, String checkinAliases, LocalizedOperationResult result) {
+        if (enabled.isTrue()) {
+            enable(Arrays.asList(user.getName()), result);
+        }
+
+        if (enabled.isFalse()) {
+            disable(Arrays.asList(user.getName()), result);
+        }
+
+        user = findUserByName(user.getName());
```

Comment:
We're having reload the user because the enable/disable happens in a transaction of its own, the other updates happen in a different transaction.

I'm happy to hear how to fix this properly though.